metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -216,6 +224,20 @@ def get_datasource_urn_from_id(self, datasource_id):
         ).json()
         schema_name = dataset_response.get("result", {}).get("schema")
         table_name = dataset_response.get("result", {}).get("table_name")
+        if table_name and RESERVED_CHARACTERS.intersection(table_name):
+            if not self.config.reserved_characters_replacement:
+                logger.warning(
+                    f"Failed to ingest following dataset due to a reserved character {RESERVED_CHARACTERS} in table name: {schema_name}.{table_name}. "
+                    "Consider using 'reserved_characters_replacement' argument to ingest dataset"
+                )
+                return None
+
+            table_name = re.sub(
+                rf"[{''.join(RESERVED_CHARACTERS)}]",
+                self.config.reserved_characters_replacement,
+                table_name,
+            )
```

Comment:
if we use our `make_dataset_urn` helper, it automatically handles this escaping logic. It also has the reserved characters set built in, so we shouldn't need that to be configurable here

Reply:
That's nice! Thank you
It's way more simple like that

Thanks a lot for your reviews, I now have a better understanding of Datahub project